pentaho-js/src/main/javascript/cdf/components/VisualizationAPIComponent.js

```diff
@@ -25,24 +30,49 @@ define([
 
     render: function(data) {
       var domElem = this.placeholder()[0];
-      var wrapper = new VisualWrapper(domElem);
-      wrapper.data = data;
-      wrapper.visualSpec = this.getVisualSpec();
-      wrapper.update()
-        .then(_.bind(this.endExec, this), _.bind(this.failExec, this));
-    },
-  	
-    getVisualSpec: function() {
-      var visualSpec = {};
-      
-      _.each(this.vizOptions, function(option) {
-        visualSpec[option[0]] = this.getParameterValue(option[1]);
-      }, this.dashboard);
+      var contextVars = new GlobalContextVars({application: "pentaho-cdf"});
+
+      var _context = new Context(contextVars);
+      var me = this;
+      _context.getAsync(this.vizId)
+          .then( function (Model) {
+            var createdModel = new Model({});
+            createdModel.data = new Table(data);
+            createdModel.height= me.height;
+            createdModel.width = me.width;
+
+            $.each(me.vizOptions, function(i, v) {
+              var prop = createdModel.get(v[0]);
+              var value = me.dashboard.getParameterValue(v[1]);
 
-      visualSpec.type = this.vizId;
+              if (prop.type.isSimple) {
```

Comment:
This conversion scheme causes me chills :-D !
To be the most transparent as possible, we would require the values of parameters to be, directly, whatever the model accepts as inputs. This is the "spec" format (or direct instances, but that's not a common use case).
In the case of visual roles, unfortunately, this would require a value like: `{attributes: ["Country"]}` — it doesn't accept an array, directly.

If we keep this special syntax for visual roles, then, instead of using the `isSimple` test below, we'd be safer testing that the property actually represents a visual role. This way, other complex typed properties could still be used.

The way to test if a property is a visual role property is to test if its `type` attribute is a subtype of `pentaho/visual/role/mapping`:

```js
var mappingType = _context.get("pentaho/visual/role/mapping").type; // cache this, somewhere above.
if(!mappingType.isSubtypeOf(prop.type)) {
  createdModel.set(propName, value);
} else {
  createdModel.get(propName).attributes.set(value);
}
```